### ISSUE TITLE
Desktop Client crashes when joining my .NET Headless

### ISSUE THREAD

**Issue by troyBORG** [Yellow-Dog-Man/Resonite-Issues#2589]
### Describe the bug?

Whenever I try to join a world hosted now by my headless my game will completely freeze.  
I can join other worlds just fine, and people can join my headless world.

A few times I was able to join my headless world, only after joining another world.  But seems random.

### To Reproduce

Attempt to join a world hosted by my `troyBORG_Headless` account.  As soon as you try to join you desktop client will freeze.

### Expected behavior

Not crash the game like it does when I try to join anyone else world.

### Screenshots / Videos

https://github.com/user-attachments/assets/6a376adc-a4e2-4ab6-8cea-9e4448583520

### Resonite Version Number

2024.7.15.1359

### What Platforms does this occur on?

Windows, Linux

### What headset if any do you use?

Quest Pro

### Log Files

Headless Log
[ResoniteServer-2024.7.15.1359-2024-07-16-03_16_39.log](https://github.com/user-attachments/files/16248035/ResoniteServer-2024.7.15.1359-2024-07-16-03_16_39.log)

Desktop Log
[DESKTOP-H0FLKOJ - 2024.7.15.1359 - 2024-07-16 03_16_53.log](https://github.com/user-attachments/files/16248032/DESKTOP-H0FLKOJ.-.2024.7.15.1359.-.2024-07-16.03_16_53.log)

**Comment by Frooxius** [comment]
What Linux system are you running the headless on?

It looks like the Brotli native library is failing to load on it. Might be due to incompatible glibc version.

**Comment by troyBORG** [comment]
Ubuntu 22.04.4 LTS



Ok.  Well here is one weird thing trying to troubleshoot my issue...

I booted the game, joined Konto's .net8 headless.
Then tried joining my Gridspace on my headless, AND IT WORKED.

I left them both and joined Konto's world again.  Then joined the Avatar Station my headless was running, and IT WORKED.

I close all the worlds.  Then joined my Gridspace, no issues, then joined then Avatar Station world..  And GAME FROZE...
[DESKTOP-H0FLKOJ - 2024.7.15.1359 - 2024-07-16 12_08_38.log](https://github.com/user-attachments/files/16254365/DESKTOP-H0FLKOJ.-.2024.7.15.1359.-.2024-07-16.12_08_38.log)

**Comment by troyBORG** [comment]
Checking.  It does say that I installed these updates on the 16th.  So unless these caused issues?

![image](https://github.com/user-attachments/assets/05a0dd96-2a43-4c53-8daf-b90074ac9524)

**Comment by troyBORG** [comment]
Here is another weird thing if this helps.  Here is a video recording, and the Desktop Logs..
https://github.com/user-attachments/assets/e3caffa1-cbe4-4ea4-84cd-2e6ff79a3cba

[DESKTOP-H0FLKOJ - 2024.7.15.1359 - 2024-07-17 02_05_07.log](https://github.com/user-attachments/files/16288963/DESKTOP-H0FLKOJ.-.2024.7.15.1359.-.2024-07-17.02_05_07.log)

[DESKTOP-H0FLKOJ - 2024.7.15.1359 - 2024-07-17 02_07_40.log](https://github.com/user-attachments/files/16288962/DESKTOP-H0FLKOJ.-.2024.7.15.1359.-.2024-07-17.02_07_40.log)

**Comment by shiftyscales** [comment]
> So it seems like the issue is just me?

Do you have any other PCs you could attempt to make the connection with @troyBORG? If not, could you try isolating other conditions to see if you can determine what is responsible? E.g. try joining while signed out / signed in to a different account, switch which account is the headless host / graphical client, etc?

Does the client crash still occur if you open a new local world before attempting to join the headless through your contacts list?

If you'd be able to rule out as many variables as possible, e.g. try and get others to replicate the issue on their own headless it could help determine where the issue is / what can be done to resolve it.

Other details that could be useful to know would include if it is specifically just that one particular world, or if you have the same issue with all sessions hosted in that specific configuration? If not, what is different about that world in particular.

The more variables you can isolate, the clearer the issue becomes.

**Comment by troyBORG** [comment]
I was going to try on my steam deck to see if it had same issue, but once the Cloud comes back up so headless can come online I'll try.

**Comment by troyBORG** [comment]
Ok.  Looks my desktop is still crashing on latest 2024.7.19.1211 build.   

My steam deck logged into my account was able to join the world without crashing...
A windows virtual machine running on my server was able to login and loaded it no issues.

So this seems like a bug local to my system..

**Comment by troyBORG** [comment]
Ok.  More testing, I told the headless to load up as the old mono headless.  Still crashing.

It seems to do it right after the LNL Relay connection would try to connect...

**Comment by troyBORG** [comment]
OHHH  I just had a discovery.  I was able to join my headless if I use the "Always use relay to connect" option.

**Comment by troyBORG** [comment]
Ok. I think I was able to resolve the issue?
I was talking to @darkysabertooth yesterday and it was suggested to specify a port in my Config.json
I tried that and it seemed like it connected without any issues...
But I reverted the config back to null and its still not freezing anymore...


Or you fixed it in `2024.7.20.94` or `2024.7.20.1115` ?   

I did also reboot my PC,  but I swore I rebooted my desktop a few times over the last week, and it didn't resolve the issue.

I'll keep it running as `null` for the port and try to see if I have any other issues.
